### PR TITLE
new: add get_system_font() - searches for common system fonts

### PR DESCRIPTION
**Pull request by WhyPenguins**
# Description

This PR just adds a new function to SplashKit, `get_system_font()`, that can be used to quickly get a nice font. It searches for common fonts across different platforms, and just returns the first one it finds, caching this result so it doesn't have to search upon subsequent calls.

## Type of change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation (update or new)

## How Has This Been Tested?

This has been tested on Windows only, and it can be confirmed that the `meiryo.ttc` font present on my computer is correctly found. Other font names have been verified by checking online, but no verification on a live system has happened - I'd be really grateful if someone set up for this was able to test it across Linux and Mac 😃 

## Testing Checklist

- [x] Tested with sktest (no effect)
- [x] Tested with skunit_tests (no effect)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have requested a review from ... on the Pull Request

### CODE REVIEW

**Comment by WhyPenguins** [comment]
[Add function to search for usable system font](https://trello.com/c/WSiBSU3V/602-add-function-to-search-for-usable-system-font)

File: coresdk/src/coresdk/text.cpp [hunk 1]
```diff
@@ -207,6 +207,59 @@ namespace splashkit_lib
...
+                std::string file_path3 = path_to_resource(fontName + ".ttf", FONT_RESOURCE);
+
+                if (
+                    !file_exists(file_path) &&
+                    !file_exists(file_path2) &&
+                    !file_exists(file_path3)
+                )
+                    continue;
+
+                font result = sk_load_font(file_path.c_str(), 64);
```
Comment (hugglesfox) [inline coresdk/src/coresdk/text.cpp:248]: I'm curious as to why `file_path2` and `file_path3` are checked if `file_path` is the only one which ends up being used.

**Review by hugglesfox — CHANGES_REQUESTED** [review]
Couple of questions but otherwise looks good.

File: coresdk/src/coresdk/text.cpp [hunk 2]
```diff
@@ -207,6 +207,59 @@ namespace splashkit_lib
...
+    font get_system_font()
+    {
+        // record if we've tried to find a system font
+        bool system_font_attempted_load = false;
+        // the system font we found (if successfull)
+        static font system_font = nullptr;
+
+        // attempt to find a common font that looks nice.
+        // This way most computers will have nice text at least
+        if (!system_font_attempted_load)
```
[the remark this answers is not in the document]
Comment (WhyPenguins) [reply coresdk/src/coresdk/text.cpp:219]: Ah, that's also meant to be static - made a few mistakes when migrating it into this file it seems...

File: coresdk/src/coresdk/text.cpp [hunk 3]
```diff
@@ -207,6 +207,59 @@ namespace splashkit_lib
...
+        {
+            system_font_attempted_load = true;
+
+            std::vector<std::string> commonFonts = {
```
[the remark this answers is not in the document]
Comment (WhyPenguins) [reply coresdk/src/coresdk/text.cpp:223]: I think this should be okay once `system_font_attempted_load` is static - this logic should only ever run once.

File: coresdk/src/coresdk/text.cpp (hunk 1 above)
Comment (WhyPenguins) [reply coresdk/src/coresdk/text.cpp:248]: Another embarrassing mistake 😓 - indeed it should be using the path that it actually found the font exists at.

**Comment by WhyPenguins** [comment]
@hugglesfox Have fixed the logic, and I think the code should be a bit cleaner now too. Thanks if you can take another look, let me know if you spot any more issues!

**Review by hugglesfox — APPROVED** [review]
lgtm!

**Comment by WhyPenguins** [comment]
Just going to tidy this up since the functionality has been merged upstream 😄 (see [here](https://github.com/splashkit/splashkit-core/pull/176))

**Pull request closed without merging**